app/models/member.rb

```diff
@@ -49,4 +52,12 @@ def send_to_ak
     )
   end
 
+  def is_donor?
+    actions.select(&:donation).present?
+  end
+
+  def is_recurring_donor?
+    actions.select{ |a| a.donation && a.form_data['is_subscription'] }.present?
```

Comment:
I'm not totally comfortable with these additions. This could noticeably affect performance in the future. Can you think about some sort of caching for this? What about a couple of flags on the member record that get updated whenever the member takes an action?